### PR TITLE
🏗 Build Spider: Los Angeles Board of Education

### PR DESCRIPTION
**Pull request by cruznunez**
## What's this PR do?
This PR adds a scraper for Los Angeles Unified School District Board of Education. 

## Why are we doing this?
Scraper requested from [spreadsheet](https://docs.google.com/spreadsheets/d/1-HwPQKwm56vZ8h7CEM65GazKLTS0086gmaEzEHXp7O4/edit?usp=sharing).

## Steps to manually test
1. Ensure the project is installed:
```
pipenv sync --dev
```
2. Activate the virtual env and enter the pipenv shell:
```
pipenv shell
```
3. Run the spider:
```
scrapy crawl losca_Board_of_ed -O test_output.csv
```
4. Monitor the output and ensure no errors are raised.
5. Inspect `test_output.csv` to ensure the data looks valid.
6. Ensure all tests pass
```
pytest
```

## Are there any smells or added technical debt to note?
This scraper uses the board's RSS feed. It does not scrape a typical HTML page. I used this approach because it had the most meetings on one page. The original link shows a calendar which did not have that many meetings on the page. Would need to click on the "previous" and "next month" arrows to see more events. 

Also, the scraper was not reading the `<link>` element as expected so had to parse it differently than the other info from the feed.

## Summary by CodeRabbit

- **New Features**
	- Introduced a new spider to scrape meeting information from the Los Angeles Unified School District Board of Education's RSS feed.
	- Added support for parsing meeting details including title, start and end times, and location.

- **Tests**
	- Implemented a comprehensive suite of unit tests to validate the functionality of the new spider, ensuring accurate parsing of meeting data.

### CODE REVIEW

File: tests/files/losca_Board_of_ed.html [hunk 1]
```diff
@@ -0,0 +1 @@
...ate></item><item><title>10/17/2024 10:00 AM - 1:00 PM Children and Families in Early Education Committee</title><link>https://www.lausd.org/site/Default.aspx?PageID=18628&amp;PageType=17&amp;DomainID=1057&amp;ModuleInstanceID=73805&amp;EventDateID=71878</link><pubDate>Thu, 17 Oct 2024 17:00:00 GMT</pubDate></item><item><title>10/17/2024 3:00 PM - 6:00 PM Personnel Commission Meeting – 3:00 p.m.</title><link>https://www.lausd.org/site/Default.aspx?PageID=18628&amp;PageType=17&amp;DomainID=1057&amp;ModuleInstanceID=73805&amp;EventDateID=71802</link><pubDate>Thu, 17 Oct 2024 22:00:00 GMT</pubDate></item><item><title>10/22/2024 11:00 AM - 3:00 PM Regular Board Meeting</title><link>https://www.lausd.org/site/Default.aspx?PageID=18628&amp;PageType=17&amp;DomainID=1057&amp;ModuleInstanceID=73805&amp;EventDateID=71688</link><pubDate>Tue, 22 Oct 2024 18:00:00 GMT</pubDate></item><item><title>10/23/2024 10:00 AM - 2:00 PM Regular Board Meeting, Including Closed Session</title><link>https://www.lausd.org/site/Default.aspx?PageID=18628&amp;PageType=17&amp;DomainID=1057&amp;ModuleInstanceID=73805&amp;EventDateID=71689</link><pubDate>Wed, 23 Oct 2024 17:00:00 GMT</pubDate></item></channel></rss>
```
[the remark this answers is not in the document]
Comment (cruznunez) [reply tests/files/losca_Board_of_ed.html:1]: The file structure is not in our control. We cannot change this.

File: city_scrapers/spiders/losca_Board_of_ed.py [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+        '...</title><link>https://www.lausd.org...EventDateID=73502<pubdate>...'
+        This string does not have a closing <link> tag even though the source
+        response does. This causes item.css('link') to return an empty tag.
+        We must parse link another way. Chose to use split.
+        """
+        split = item.get().split("<link>")[1]
+        link = split.split("<pubdate>")[0]
+        links = [{"title": "Meeting Details", "href": link}]
+
+        return links
```
[the remark this answers is not in the document]
Comment (cruznunez) [reply city_scrapers/spiders/losca_Board_of_ed.py]: The problem is that scrapy doesn't see the closing `</link>` tag so this code breaks. I'll add this with an improved version.

File: city_scrapers/spiders/losca_Board_of_ed.py [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+    def _parse_title(self, item):
+        """
+        Parse meeting title. RSS feed titles always start with timestamp.
+        Ex: '9/19/2024 10:00 AM - 1:00 PM Children... Early Education Committee'
+        Remove timestamp from string and return title.
+        """
+        raw = item.css("title::text").get()
+        no_stamp = raw.split()[6:-1]
+        title = " ".join(no_stamp)
+        return title
```
[the remark this answers is not in the document]
Comment (cruznunez) [reply city_scrapers/spiders/losca_Board_of_ed.py]: This probably makes it harder to read but okay.

File: city_scrapers/spiders/losca_Board_of_ed.py [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+
+    def _parse_end(self, item):
+        """
+        Parse end datetime as a naive datetime object.
+        End time is in title.
+        """
+        raw = item.css("title::text").get().split()
+        date = raw[0]
+        time = " ".join(raw[4:6])
+        return parse(f"{date} {time}")
```
[the remark this answers is not in the document]
Comment (cruznunez) [reply city_scrapers/spiders/losca_Board_of_ed.py]: Turned 4 LOC to 10. Nice.

**Review by lamle-ea — APPROVED** [review]
LGTM

**Pull request merged**